core/domain/state_domain.py

```diff
@@ -1568,15 +1568,28 @@ def is_rte_content_supported_on_android(self):
         Returns:
             bool. Whether the RTE components in the state is valid.
         """
-        regex_string = r'oppia-noninteractive-'
-        regex_string += (
-            '|'.join(android_validation_constants.INVALID_RTE_COMPONENTS))
-        regex_string = regex_string[:-1]
-        regex = re.compile(regex_string)
-        if self.content and regex.search(self.content.html):
+        def whitelist_checker(html):
```

Comment:
Always start function names with verbs. require_rte_components_to_match_whitelist or something like that. Or perhaps better, _validate_for_android

Reply:
Hm, I'm not sure about using a private function name because I'm passing this on to another function.

Reply:
what about require_component_names_to_match?

Reply:
OK, doesn't need to be private then.

That sounds fine too.